File: src/main/java/com/nkming/powermenu/LauncherActivity.java
/*
 * LauncherActivity.java
 *
 * Author: Ming Tsang
 * Copyright (c) 2015 Ming Tsang
 * Refer to LICENSE for details
 */

package com.nkming.powermenu;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;

public class LauncherActivity extends Activity
{
	@Override
	public void onCreate(Bundle savedInstanceState)
	{
		super.onCreate(savedInstanceState);
		startActivity(new Intent(this, MainActivity.class));
		finish();
	}
}

File: src/main/kotlin/com/nkming/powermenu/LauncherActivity.kt
package com.nkming.powermenu

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

/**
 * This class is a proxy class such that the launcher icon could be disabled
 */
class LauncherActivity : AppCompatActivity()
{
	override fun onCreate(savedInstanceState: Bundle?)
	{
		super.onCreate(savedInstanceState)
		startActivity(Intent(this, MainActivity::class.java))
		finish()
	}
}
